server/py/services/api/crud/model_monitoring/model_endpoints.py

```diff
@@ -52,11 +52,12 @@
 class ModelEndpoints:
     """Provide different methods for handling model endpoints such as listing, writing and deleting"""
 
-    def create_model_endpoint(
+    async def create_model_endpoint(
         self,
         db_session: sqlalchemy.orm.Session,
         model_endpoint: mlrun.common.schemas.ModelEndpoint,
         creation_strategy: mlrun.common.schemas.ModelEndpointCreationStrategy,
+        model_path: Optional[str] = None,
     ) -> mlrun.common.schemas.ModelEndpoint:
```

Comment:
```suggestion
    ) -> Union[mlrun.common.schemas.ModelEndpoint, None]:
```

Reply:
That would be
```suggestion
    ) -> Optional[mlrun.common.schemas.ModelEndpoint]:
```
But more importantly, all callers need to be ready to handle `None`, and their own type hints need to be adjusted accordingly.